File: Bin2Dec/Bin2Dec.java
// Online Java Compiler
// Use this editor to write, compile and run your Java code online
import java.util.Scanner;

class HelloWorld {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        String binaryInput;
        int decimalOutput = 0;

        System.out.print("Enter a binary number up to 8 digits: ");
        binaryInput = input.nextLine();
        
         if(binaryInput.length() > 8) {
            System.out.println("Error: Only 8 digits are allowed.");
            return;
        }
        
        System.out.println(convertBin2Dec(Integer.parseInt(binaryInput)));
    }
    
    public static int convertBin2Dec(int binaryInput){
        int n=0;
        int decimal=0;
        while(true){
            if(binaryInput==0){
                break;
            }else{
                int temp=binaryInput%10;
                decimal+=temp * Math.pow(2, n);
                binaryInput=binaryInput/10;
                n++;
            }
        }
        return decimal;
    }
}

File: Bin2Dec/Bin2Dec.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println("Hello, Coders")
	fmt.Println(bin2Dec(1000))

}

func bin2Dec(input int) (decimal int) {
	var n int = 0
	decimal = 0
	for true {
		fmt.Println("In loop")
		if input == 0 {
			break
		} else {
			var temp int = input % 10
			decimal = decimal + temp*int(math.Pow(2, float64(n)))
			input = input / 10
		}
		n = n + 1
	}
	fmt.Println("printing things", decimal)
	return
}
